mmcv/transform/processing.py

```diff
@@ -0,0 +1,382 @@
+        results['width'] = img_shape[1]
+
+        # crop the gt_semantic_seg
+        if 'gt_semantic_seg' in results:
+            results['gt_semantic_seg'] = mmcv.imcrop(
+                results['gt_semantic_seg'], bboxes=np.array([x1, y1, x2, y2]))
+
+        results['pad_shape'] = img_shape
+        return results
+
+    def __repr__(self):
+        repr_str = self.__class__.__name__
+        repr_str += f', crop_size = {self.crop_size}'
+        repr_str += f', pad_val = {self.pad_val}'
+        repr_str += f', pad_mode = {self.pad_mode}'
+        return repr_str
+
+
+@TRANSFORMS.register_module()
+class RandomGrayscale(BaseTransform):
+    """Randomly convert image to grayscale with a probability of prob.
+
+    Args:
+        prob (float, optional): Probability that image should be converted to
+            grayscale. Default: 0.1.. Defaults to 0.1.
+        keep_channel (bool, optional): Whether keep channel number the same as
+            input. Defaults to False.
+        channel_weights (tuple, optional): Channel weights to compute gray
+            image. Defaults to (1., 1., 1.).
+        color_format (str, optional): Color format set to be any of 'bgr',
+            'rgb', 'hsv' . Defaults to 'bgr'.
+    """
+
+    def __init__(self,
+                 prob: float = 0.1,
+                 keep_channel: bool = False,
+                 channel_weights: Sequence[float] = (1., 1., 1.),
+                 color_format: str = 'bgr'):
+        super().__init__()
+        self.prob = prob
+        self.keep_channel = keep_channel
+        self.channel_weights = channel_weights
+        assert color_format in ['bgr', 'rgb', 'hsv']
+        self.color_format = color_format
+
+    def transform(self, results: dict) -> dict:
+        """Apply random grayscale on results.
+
+        Args:
+            results (dict): Result dict contains the data to transform.
+
+        Returns:
+           dict[str: list]: The augmented data, where each value is wrapped
+               into a list.
+        """
+        img = results['img']
+        # convert hsv to bgr
+        if self.color_format == 'hsv':
+            img = mmcv.hsv2bgr(img)
+        num_output_channels = img.shape[2]
+        if random.random() < self.prob:
+            if num_output_channels > 1:
+                assert num_output_channels == len(self.channel_weights)
+                normalized_weights = np.array(self.channel_weights) / sum(
+                    self.channel_weights)
+                img = sum([
+                    weight * img[:, :, i]
+                    for i, weight in enumerate(normalized_weights)
+                ])
+                if self.keep_channel:
+                    img = img[:, :, None]
+                    results['img'] = np.dstack(
+                        [img for _ in range(num_output_channels)])
+                else:
+                    results['img'] = img
+                return results
+        results['img'] = img
```

Comment:
should we revert color model if original image is hsv

Reply:
My bad. HSV image will be converted to BGR format first.